Aspire Budgeting/Facilities/AppCoordinator.swift

```diff
@@ -0,0 +1,142 @@
+  func verify(file: File) {
+    //pass default file to sheets manager
+    print(file)
+  }
+
+  private let objectCreator: ObjectCreator
+  private let stateManager: AppStateManager
+  private let localAuthManager: LocalAuthorizationManager
+  private let userManager: AppUserManager
+  private let driveManager: GoogleDriveManager
+
+  private let defaultsSheetsKey = "Aspire_Sheet"
+
+  private var stateManagerSink: AnyCancellable!
+  private var driveManagerSink: AnyCancellable!
+
+  private var userDefaults: AspireUserDefaults
+
+  private(set) var defaultFile: File?
+
+  @Published private(set) var fileSelectorViewModel: FileSelectorViewModel?
+
+  init(objectCreator: ObjectCreator,
+       userDefaults: AspireUserDefaults = UserDefaults.standard) {
+    self.objectCreator = objectCreator
+    self.stateManager = objectCreator.stateManager
+    self.localAuthManager = objectCreator.localAuthorizationManager
+    self.userManager = objectCreator.userManager
+    self.driveManager = objectCreator.driveManager
+
+    self.userDefaults = userDefaults
+  }
+
+  func checkForDefaultFile() {
+    guard let data = userDefaults.data(forKey: defaultsSheetsKey),
+      let file = try? JSONDecoder().decode(File.self, from: data) else {
+      os_log(
+        "No default Google Sheet found",
+        log: .appCoordinator,
+        type: .default
+      )
+      return
+    }
+
+    os_log(
+      "Default Google Sheet found.",
+      log: .appCoordinator,
+      type: .default
+    )
+    defaultFile = file
+    stateManager.hasDefaultFile()
+//    NotificationCenter.default.post(
+//      name: .hasSheetInDefaults,
+//      object: nil,
+//      userInfo: [Notification.Name.hasSheetInDefaults: file]
+//    )
+  }
+
+  private func handle(currentState: AppState) {
+    switch currentState {
+    case .loggedOut:
+      userManager.authenticateExternally()
+
+    case .authenticatedExternally:
+      localAuthManager.authenticateLocally()
+
+    case .authenticatedLocally:
+      checkForDefaultFile()
+      if !hasDefaultFile {
+        driveManager.getFileList()
+      }
+
+    default:
+      print("The current state is \(currentState)")
+    }
+  }
+}
+
+// MARK: Computed Properties
+extension AppCoordinator {
+  var isLoggedOut: Bool {
+    stateManager.isLoggedOut
+  }
+
+  var needsLocalAuth: Bool {
+    stateManager.isLocalAuthNeeded
+  }
+
+  var hasDefaultFile: Bool {
+    defaultFile != nil
+  }
+}
+
+// MARK: App Life Cycle
+extension AppCoordinator {
+  func start() {
+    stateManagerSink =
+      stateManager
+      .currentStatePublisher
+      .sink { [self] currentState in
+        self.handle(currentState: currentState)
+      }
+
+//    filesListSink = driveManager
+//      .$fileList
+//      .filter { $0 != nil }
+//      .map { $0! }
+//      .sink {
+//        self.fileSelectorViewModel = FileSelectorViewModel(files: $0, delegate: self)
+//      }
+
+    driveManagerSink = driveManager.$state.sink(receiveValue: { state in
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/02d5375236d0328ac5fc8dbb7c000406514d930c/Aspire Budgeting/Facilities/AppCoordinator.swift#L124"></span>Trailing closure syntax should be used whenever possible.<br />
<code>trailing_closure</code> <code>AppCoordinator.swift:124</code></td>
    </tr>
  </tbody>
</table>